plugins/modules/dcnm_template.py

```diff
@@ -0,0 +1,629 @@
+      config:
+        - name: template_101
+          description: "Template_101"
+          tags: "internal policy 101"
+          content: |
+            telemetry
+              certificate /bootflash/telegraf.crt telegraf
+              destination-profile
+                use-vrf management
+              destination-group 101
+                ip address 10.195.225.176 port 57101 protocol gRPC encoding GPB
+              sensor-group 101
+                data-source DME
+                path sys/ch depth unbounded
+              subscription 101
+                dst-grp 101
+                snsr-grp 101 sample-interval 10101
+
+        - name: template_102
+          description: "Template_102"
+          tags: "internal policy 102"
+          content: |
+            telemetry
+              certificate /bootflash/telegraf.crt telegraf
+              destination-profile
+                use-vrf management
+              destination-group 1
+                ip address 10.195.225.102 port 57102 protocol gRPC encoding GPB
+              sensor-group 102
+                data-source DME
+                path sys/ch depth unbounded
+              subscription 102
+                dst-grp 102
+                snsr-grp 102 sample-interval 10102
+
+# To delete templates 
+
+- name: Delete templates
+    cisco.dcnm.dcnm_template:
+      state: deleted       # only choose form [merged, deleted, query]
+      config:
+        - name: template_101
+
+        - name: template_102
+
+        - name: template_103
+
+        - name: template_104
+
+# To query templates 
+
+- name: Query templates
+    cisco.dcnm.dcnm_template:
+      state: query       # only choose form [merged, deleted, query]
+      config:
+        - name: template_101
+
+        - name: template_102
+
+        - name: template_103
+
+        - name: template_104
+'''
+
+import time
+import json
+import re
+import copy
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.connection import Connection
+from ansible_collections.cisco.dcnm.plugins.module_utils.network.dcnm.dcnm import \
+    dcnm_send, validate_list_of_dicts 
+
+import datetime
+
+class DcnmTemplate:
+
+    def __init__(self, module):
+        self.module        = module
+        self.params        = module.params
+        self.config        = copy.deepcopy(module.params.get('config'))
+        self.want          = []
+        self.have          = []
+        self.pb_input      = []
+        self.diff_create   = []
+        self.diff_delete   = []
+        self.diff_query    = []
+        self.valid_fail    = []
+        self.template_info = []
+        self.fd            = None
+        self.changed_dict  = [{'merged' : [], 'deleted' : [], 'query' : [], 'failed' : []}]
+
+        self.result = dict(
+            changed  = False,
+            diff     = [],
+            response = []
+        )
+
+    def log_msg (self, msg):
```

Comment:
Same question here for `log_msg`

Reply:
I just left the function definition alone, in case we need this for debugging later. This will not be called in the final version.